### ISSUE TITLE
Dynamic arrays

### ISSUE THREAD

**Issue by MaxGraey** [dcodeIO/AssemblyScript#97]
As you know I try to implement resizeable arrays and its methods but I has some problems need some advisements.

```unsafeGrow``` return new pointer to reallocated memory, but current Array's implementation hasn't aggregated pointer to this memory. Header (length and capacity) store in the same place as array's data in optimisation reason as I understand. After **unsafeGrow** we need somehow update our pointer (**this** pointer) to instance of array but this is not possible. One possible solution for resolving this - is make pointer to data in aggregated style, but in this case we need implicit access to elements via **index operator** or hardcoded this in compiler routine. What do you think?

**Comment by dcodeIO** [comment]
Related: https://github.com/dcodeIO/AssemblyScript/issues/88

**Comment by MaxGraey** [comment]
Unfortunately **unsafeGrow** not work as expected

**Comment by dcodeIO** [comment]
Yeah, I didn't double check my implementation once I figured that it doesn't make sense anyway.

**Comment by dcodeIO** [comment]
Should be possible now to implement push etc :)

**Comment by MaxGraey** [comment]
Thanks! I will try to implement Array's dynamic functionality ASAP.
